docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/2.deploy.md

```diff
@@ -0,0 +1,95 @@
+# Deploy full-text index
+
+Nebula Graph full-text indexes are powered by [Elasticserch](https://en.wikipedia.org/wiki/Elasticsearch). Full-text indexes are managed through built-in procedures. They can be created only for variable `STRING` and `FIXED_STRING` properties when the listener cluster and the elastic search cluster are ready.
+
+## Deploy Elasticsearch cluster
+
+To deploy an Elasticsearch cluster, see the [Elasticsearch documentation](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-deploy-elasticsearch.html).
+
+When the Elasticsearch cluster is started, add the index template for the Nebula Graph full-text. For example:
+
+```json
+{
+ "template": "nebula*",
+  "settings": {
+    "index": {
+      "number_of_shards": 3,
+      "number_of_replicas": 1
+    }
+  },
+  "mappings": {
+    "properties" : {
+            "tag_id" : { "type" : "long" },
+            "column_id" : { "type" : "text" },
+            "value" :{ "type" : "text"}
+        }
+  }
+}
+```
+
+Make sure that you specify the following fields in strict accordance with the preceding template format:
+
+```text
+"template": "nebula*"
+"tag_id" : { "type" : "long" },
+"column_id" : { "type" : "text" },
+"value" :{ "type" : "text"}
+```
+
+## Deploy full-text cluster
+
+Full-text index data is wrote to the Elasticsearch cluster asynchronously. To use the full-text index search, you must run the full-text storage services as Listeners. Configure the `listener_path` parameter for the Listeners. For example:
```

Comment:
you must run one or more Nebula storage process as raft-listener, which will write the text into ElasticSearch cluster.